src/main/java/use_cases/add_edit_category_use_case/CategoryID.java

```diff
@@ -0,0 +1,51 @@
+package use_cases.add_edit_category_use_case;
+
+import entities.Category;
+import entities.SessionStorage;
+
+public class CategoryID {
```

Comment:
Is this the input data class that was gonna be split into two for add vs. edit? If so, maybe you could keep this singular one and distinguish between the two uses by setting old_category to null when it's creating a new one, and then you just check for null in the interactor to see what to do. Do you think that could work?

Reply:
I agree. I think in that case, she just need to create a new constructor with out the oldCategory variable.

Reply:
This is the same comment I had in AddCategoryV, I'm copying it here since they're related!

That null is not a user input. If you check AddCategoryV class, the null is an assignment. It is just so the CategoryC can call both types of use cases with the same method CategoryC.categoryInMonth(). (This is an accommodation to what Ari suggested about combining the input data!)

public AddCategoryV(CategoryC controller,int monthID, SessionStorage currSession): this is the constructor of AddCategoryV so don't worry.

Reply:
Resolved: variable assigned null in view.